pkg/client/options.go

```diff
@@ -72,6 +72,14 @@ func IdleConnTimeout(timeout time.Duration) Option {
 	}
 }
 
+// HttpClient specifies the http.Client to use. If provided the rest of the connection options are ignored.
+func HttpClient(c *http.Client) Option {
```

Comment:
small nit (ignore!) for readability, better have acronyms all upper case. e.g. YAMLNode, JSONText, HTTPClient, FTPClient.